kiosk/kiosk_browser/browser_widget.py

```diff
@@ -1,13 +1,19 @@
 from PyQt5.QtCore import QTimer
 from PyQt5.QtWidgets import QShortcut
-from PyQt5.QtWebEngineWidgets import QWebEngineView, QWebEnginePage
+from PyQt5.QtWebEngineWidgets import QWebEngineView, QWebEnginePage, QWebEngineProfile
 from PyQt5.QtWidgets import QSizePolicy
 
 class BrowserWidget(QWebEngineView):
-    def __init__(self, url, *args, **kwargs):
+
+    def __init__(self, name, url, *args, **kwargs):
         QWebEngineView.__init__(self, *args, **kwargs)
 
-        self.load(url)
+        profile = QWebEngineProfile.defaultProfile()
+        profile.setHttpUserAgent(f"{name} {profile.httpUserAgent()}")
```

Comment:
I think it would be nice to stick to the [conventional format](https://en.wikipedia.org/wiki/User_agent#Format_for_human-operated_web_browsers) here, which is something like `Mozilla/5.0 (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Mobile/7B405`. There is also a panoply of user agent strings using this format in the [platform.js test suite](https://github.com/bestiejs/platform.js/blob/master/test/test.js).

This implies replacing the contents of the first set of parentheses (which is "Details of the system in which the browser is running") with something like `(X11; PlayOS 19.8.0)`.

Reply:
I chose to kept the Linux part of the system, that’s not bothering and it adds extra information.

`Mozilla/5.0 (X11; Linux x86_64; PlayOS 2.1.0) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.0 Chrome/69.0.3497.128 Safari/537.36`

I added a fallback in case there is something wrong with the matching, in this case, the PlayOS version is set at the begining of the user agent.

Reply:
Memory is not freed when recreating a new page instance each time:

![memory-profile-new-page-instance](https://user-images.githubusercontent.com/6768842/65340979-a2598600-dbcf-11e9-91cd-584683652024.png)

No memory leak when using the same page instance:

![memory-profile-same-page-instance](https://user-images.githubusercontent.com/6768842/65340981-a2598600-dbcf-11e9-9814-0fdddec97693.png)